executor/builder.go

```diff
@@ -5102,6 +5111,31 @@ func partitionPruning(ctx sessionctx.Context, tbl table.PartitionedTable, conds
 	return ret, nil
 }
 
+func getPartitionIdsAfterPruning(ctx sessionctx.Context, tbl table.PartitionedTable, conds []expression.Expression, partitionNames []model.CIStr,
+	columns []*expression.Column, columnNames types.NameSlice) (map[int64]struct{}, error) {
```

Comment:
```suggestion
func getPartitionIdsAfterPruning(ctx sessionctx.Context, tbl table.Table, partInfo PartitionInfo) (map[int64]struct{}, error) {
```
Then you can check if it is a `table.PartitionedTable` etc. in the function and simplify the caller code.